CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/ParameterInputPane.java

```diff
@@ -0,0 +1,166 @@
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.gui;
+
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameter;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameterListener;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameterType;
+import au.gov.asd.tac.constellation.plugins.parameters.types.ParameterValue;
+import au.gov.asd.tac.constellation.utilities.gui.field.framework.ConstellationInput;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.HBox;
+import au.gov.asd.tac.constellation.utilities.gui.field.framework.ConstellationInputListener;
+import au.gov.asd.tac.constellation.utilities.gui.field.framework.InputValidator;
+
+/**
+ * An abstract base class for {@link ParamerPanes} that get user input through a {@link InputField} 
+ * to modify {@link PluginParameter} objects in Constellation.
+ * This class and it's extended classes act as the link between {@link PluignParameter} objects
+ * and {@link ConstellationInput} objects.
+ * 
+ * This class provides standard functionality for this link and defines some implementation specific 
+ * methods to assist in the nuanced interactions across different {@link PluginParameterType} implementations.
+ * 
+ * Visually, extensions of this class are represented by a {@link HBox} which contains a 
+ * {@link ConstellationInput}. This {@link HBox} is typically added to a {@link GridPane} by
+ * the {@link PluginParametersPane}.
+ * 
+ * Visual examples of the {@link ParameterInputPane} can be found in the DataAccessView or the SphereGraphBuilder.
+ * 
+ * @author capricornunicorn123
+ * @param <T> the Value represented by the {@link PluginParameter}
+ * @param <V> the Value represented by the {@link ConstellationInput}
+ */
+public abstract class ParameterInputPane<T extends ParameterValue, V extends Object> extends HBox {
+    
+    public final ConstellationInput<V> input;
+    public final PluginParameter<T> parameter;
+    
+    ParameterInputPane(final ConstellationInput<V> input, final PluginParameter<T> parameter){
+        
+        //Set local references to the PluginParameter and the ConstellationInput
+        this.input = input;
+        this.parameter = parameter;
+        
+        //Ensure the input pane is correctly vsible and enabled
+        updateFieldEnablement();
+        updateFieldVisibility();
+
+        //Update the prompt text of the input
+        this.input.setPromptText(parameter.getDescription());
+        
+        // Ensure that the input and the parameter are listening to changes on each other
+        this.input.addListener(getFieldChangeListener(parameter));
+        parameter.addListener(getPluginParameterListener());
+        
+        // Add a validator that uses the parameter validation to validae the value of an input.
+        this.input.addValidator((String s) -> {
+            return parameter.validateString(s);
```

Comment:
make parameter final or remove explicit type. I also believe you should be able to omit the curly braces and return keyword here